logbucket/logbucket.go

```diff
@@ -234,6 +234,12 @@ func (d *Downloader) accessLogBucketPageCallback(processedObjects map[string]tim
 		}
 
 		if time.Since(*obj.LastModified) < d.BackfillInterval {
+			if err := d.SetProcessed(*obj.Key); err != nil {
+				logrus.Error("Error setting state of object as processed: %s", err)
```

Comment:
Should we exit here if we fail to write the object as processed? If there's a race and two or more clients try to claim the same object, Dynamo should fail the write on all but one (since we check for existing S3Object column here https://github.com/honeycombio/honeyaws/blob/master/state/state.go#L141). In that case, the winner of the race would download the object, and all other clients would move on to other files.

Reply:
Ah yeah! Great catch.